assets/source/tests/custom-matchers.js

```diff
@@ -0,0 +1,49 @@
+expect.extend( {
+	/**
+	 * Custom matcher to check the presence and optionally the value of a search query param in the given location.
+	 *
+	 * @param {Location} receivedLocation Location to be checked, for example `window.location`.
+	 * @param {string} param Parameter to check the presence of.
+	 * @param {string} [value] Value of the parameter to be checked.
+	 * @return {{message: string, pass: boolean}} Jest compatible matcher output.
+	 */
+	toContainURLSearchParam( receivedLocation, param, value ) {
+		const hasValue = arguments.length === 3;
```

Comment:
💅 

I guess this for allowing falsy values like null or undefined as the expected param value?

Reply:
Yes, to allow "expect the location to have URL param `path` of `/some/thing`" as well as just  "expect the location to have URL param `redirect`". This is to follow the convention of [`toHaveProperty`](https://jestjs.io/docs/expect#tohavepropertykeypath-value)

> Custom matcher to check **the presence and optionally the value** of a search query param in the given location.

Do you think it could be stated more clearly?

Reply:
Was more about 

`const hasValue = arguments.length === 3;` instead of just checking `value` prop directly.